lib/util/mp4_box_parsers.js

```diff
@@ -0,0 +1,296 @@
+/*! @license
+ * Shaka Player
+ * Copyright 2016 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+goog.provide('shaka.util.Mp4BoxParsers');
+
+goog.require('shaka.util.DataViewReader');
+goog.require('shaka.util.Iterables');
+
+shaka.util.Mp4BoxParsers = class {
+  /**
+   * Parses a TFHD Box.
+   * @param {!shaka.util.DataViewReader} reader
+   * @param {!number} flags
+   * @return {!shaka.util.ParsedTFHDBox}
+   */
+  static parseTFHD(reader, flags) {
+    let defaultSampleDuration = null;
+
+    reader.skip(4); // Skip "track_ID"
+
+    // Skip "base_data_offset" if present.
+    if (flags & 0x000001) {
+      reader.skip(8);
+    }
+
+    // Skip "sample_description_index" if present.
+    if (flags & 0x000002) {
+      reader.skip(4);
+    }
+
+    // Read "default_sample_duration" if present.
+    if (flags & 0x000008) {
+      defaultSampleDuration = reader.readUint32();
+    }
+
+    /** @type {!shaka.util.ParsedTFHDBox} */
+    const parsedTFHD = {
+      defaultSampleDuration: defaultSampleDuration,
+    };
+
+    return parsedTFHD;
```

Comment:
The compiler knows the return type of the method, so stylistically, you should be able to return the structure inline without the need for a local variable or extra type annotation:

```js
return {
  defaultSampleDuration: defaultSampleDuration,
};
```

Also, another style note is that we can/should use the short-hand syntax in which the key and value have the same name:

```js
return {
  defaultSampleDuration,
};
```

Same for the other box types in this file.